src/core/syscalls/business_logic_syscall_handler.rs

```diff
@@ -161,28 +180,95 @@ impl SyscallHandler for BusinessLogicSyscallHandler {
+                Some(maybe_reloc_cow) => match maybe_reloc_cow.deref() {
+                    MaybeRelocatable::Int(int) => calldata.push(int.clone()),
+                    _ => return Err(SyscallHandlerError::ExpectedMaybeRelocatableInt),
+                },
+            }
+        }
 
-        self.events.push(OrderedEvent::new(order, keys, data));
+        let mut code_address = None; // Optional[int]
+        let mut class_hash = None; // Optional[bytes]
+
+        match syscall_name {
+            "call_contract" => {
+                code_address = Some(request.contract_address);
+                let contract_address = code_address;
+                let caller_address = self.contract_address;
+                let entry_point_type = EntryPointType::External;
+                let call_type = CallType::Call;
+            }
+            "delegate_call" => {
+                code_address = Some(request.contract_address);
+                let contract_address = self.contract_address;
+                let caller_address = self.caller_address;
+                let entry_point_type = EntryPointType::External;
+                let call_type = CallType::Delegate;
+            }
+            "delegate_l1_handler" => {
+                code_address = Some(request.contract_address);
+                let contract_address = self.contract_address;
+                let caller_address = self.caller_address;
+                let entry_point_type = EntryPointType::L1Handler;
+                let call_type = CallType::Delegate;
+            }
+            "library_call" => {
+                class_hash = Some(request.class_hash.to_be_bytes());
+                let contract_address = self.contract_address;
+                let caller_address = self.caller_address;
+                let entry_point_type = EntryPointType::External;
+                let call_type = CallType::Delegate;
+            }
+            "library_call_l1_handler" => {
+                class_hash = Some(request.class_hash.to_be_bytes());
+                let contract_address = self.contract_address;
+                let caller_address = self.caller_address;
+                let entry_point_type = EntryPointType::L1Handler;
+                let call_type = CallType::Delegate;
+            }
+            _ => return Err(SyscallHandlerError::UnknownSyscall), // TODO add some message to this error.
+        }
 
-        // Update events count.
-        self.tx_execution_context.n_emitted_events += 1;
-        Ok(())
+        // let call = self.execute_entry_point(
+        //     call_type,
+        //     class_hash,
+        //     contract_address,
+        //     code_address,
+        //     request.selector,
+        //     entry_point_type,
+        //     calldata,
+        //     caller_address,
+        // )
+
+        // return self.execute_entry_point(call=call)
+
+        // TODO, remove this once used the commented code.
+        Ok(Vec::new())
```

Comment:
mmmm,
I think it is better to put a `todo!() `or an `unimplemented!()`  here instead of returning an empty value